src/qmoperators/two_electron/ExchangePotentialD1.cpp

```diff
@@ -53,9 +53,9 @@ void ExchangePotentialD1::setupInternal(double prec) {
             Orbital &phi_i = iter.orbital(0);
             int idx = iter.idx(0);
             for (int j = 0; j < Phi.size(); j++) {
+                if (mpi::my_orb(phi_i) and j <= idx) continue; // compute only i<j for own block
                 Orbital &phi_j = (*this->orbitals)[j];
-                if (idx == j) continue; // skip diagonal terms
-                if (mpi::my_orb(phi_i) and mpi::my_orb(phi_j)) calcInternal(idx, j);
+                if (mpi::my_orb(phi_j)) calcInternal(idx, j, phi_i, phi_j);
```

Comment:
Can't you use the `calc_internal(i,j)` here?

Reply:
phi_i is not readable in Phi, if it is not owner, so it must be given explicitely